packages/x-components/src/x-modules/search/store/actions/fetch-and-save-search-response.action.ts

```diff
@@ -51,6 +48,34 @@ const { fetchAndSave, cancelPrevious } = createFetchAndSaveActions<
   }
 });
 
+/**
+ * Enriches the {@link SearchRequest} object with the origin and page properties taken from the
+ * {@link SearchState | search state}.
+ *
+ * @param request - The {@link PageableSearchRequest}.
+ * @param state - {@link SearchState}.
+ *
+ * @returns The search request.
+ * @internal
+ */
+function enrichRequest(request: PageableSearchRequest, state: SearchState): SearchRequest {
+  const { page, ...restRequest } = request;
+  const {
+    config: { pageSize },
+    origin,
+    results
+  } = state;
+  const start = page === 1 ? 0 : results.length;
+
+  return {
+    ...restRequest,
+    // eslint-disable-next-line @typescript-eslint/no-extra-parens
+    ...(origin && { origin }),
+    start,
+    rows: pageSize * page! - start
```

Comment:
`page` is an optional field in `PageableSearchRequest`, but we have to tell TS that it will have value when it might not be true. 
If we'll always have `page`, it shouldn't be optional in the type declaration. If not, then we have to handle unexpected values for it. In this line and in line 68 the logic breaks if `page` is note defined

Reply:
I think page is not optional as it will be as least `1` (default value in the store)

Reply:
If it's not possible for `page` to reach this function without value then it's all good 👍

Reply:
Done